containers.go

```diff
@@ -312,22 +312,30 @@ func runContainer(daemon *docker.Client, id string, logger log15.Logger, logfile
 	// If we're the outer shell, log straight to stderr, nothing fancy
 	stdout := io.Writer(os.Stdout)
 	stream := io.Writer(os.Stderr)
+	var log, hookedR, hookedW *os.File
+	var fdsToClose []io.Closer
+	var err error
```

Comment:
Actually, if I read the code correctly, non of these pre-declares (apart from the closer slice) you added are needed here, and they just convolute the code. Please revert to the original code that declared them when used.